ompi/mpi/c/ireduce.c

```diff
@@ -135,10 +135,27 @@ int MPI_Ireduce(const void *sendbuf, void *recvbuf, int count,
         return MPI_SUCCESS;
     }
 
+    void *updated_recvbuf = recvbuf;
+    const void *updated_sendbuf = sendbuf;
+    if(OMPI_COMM_IS_INTRA(comm)) {
+        if( ompi_comm_rank(comm) != root ) {
+            updated_recvbuf = NULL;
+        }
+    } else {
+        if( MPI_ROOT == root ) {
+            updated_sendbuf = NULL;
+        } if( MPI_PROC_NULL == root ) {
```

Comment:
else missing?

Reply:
No need for the `else` case here, that would default to the initial values for `updated_recvbuf` and `updated_sendbuf`.